pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -234,23 +236,68 @@ func (a *Bootstrap) addSystemdUnits(uri string, templateData *bootstrapTemplateD
 	}
 
 	for _, childInfo := range children {
-		name := childInfo.Name()
-		file, err := data.Assets.Open(path.Join(uri, name))
+		dir := path.Join(uri, childInfo.Name())
+		file, err := data.Assets.Open(dir)
 		if err != nil {
 			return err
 		}
 		defer file.Close()
 
-		name, data, err := readFile(name, file, templateData)
+		info, err := file.Stat()
 		if err != nil {
 			return err
 		}
 
-		unit := igntypes.Unit{Name: name, Contents: string(data)}
-		if _, ok := enabled[name]; ok {
-			unit.Enabled = util.BoolToPtr(true)
+		if info.IsDir() {
+			children, err := file.Readdir(0)
+			if err != nil {
+				return err
+			}
+			file.Close()
```

Comment:
We usually ignore the `error` returned by `Close()`, because we're calling it from a `defer` and almost all the time it will succeed without errors.  But since we're calling it here without the `defer`, I think we should have:

```go
err = file.Close()
if err != nil {
  return err
}
```
here.  And Go makes me miss Python's auto-bubbling, `try`/`except` errors ;).

Reply:
If you miss those, you should check out [Rust's `?` operator](https://doc.rust-lang.org/1.30.0/book/second-edition/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator).

Reply:
> If you miss those, you should check out Rust's `?` operator.

Next week we should rewrite the installer in Rust... ;).